src/ressa/node_pattern.rs

```diff
@@ -47,13 +47,48 @@ pub struct NodePattern {
     /// Transparently forward to child nodes
     #[serde(default = "bool::default")]
     pub transparent: bool,
+
+    /// Language this node applies to
+    #[serde(default = "Option::default")]
+    pub language: Option<Language>,
 }
 
 impl NodePattern {
+    /// Determine whether this node matches the language of the provided node
+    /// If no language initialized, assume failure.
+    pub fn language_matches(&self, node: &impl NodeLanguage) -> bool {
+        if let Some(lang) = self.language {
+            lang == Language::Unknown || lang == node.get_language()
+        } else {
+            false
+        }
```

Comment:
This is another way of doing this, just showing this since you may not have seen this syntax before, not that this really needs changing or anything
```rust
match self.language {
    Some(Language::Unknown) => true,
    Some(lang) if lang == node.get_language() => true,
    _ => false,
}
```
You can also do other stuff like this to bind variants in a match arm to a variable
```rust
match self.language {
    Some(lang @ Language::Unknown) | Some(lang @ Language::Go) => {}, // do something with lang...
    other @ None => {}
}
```

Reply:
Did not know about that @ syntax. Replacing with a `match` actually does make it a bit cleaner IMO, so I've made a hybrid update.